### PR TITLE
feat: add iterator to `Mutset` and `MutMap`

### LINKED ISSUE #4263: Add `iterator` to `MutSet` and `MutMap`
Based on Stephen's work in https://github.com/flix/flix/pull/4259.

Also, I think we should add:
- `keyIterator` and `valueIterator` to Map/MutMap.

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #4263

### CODE REVIEW

**Comment by jaschdoc** [comment]
RFR

File: main/src/library/MutMap.flix [hunk 1]
```diff
@@ -577,6 +577,38 @@ namespace MutMap {
...
+    ///
+    pub def iterator(m: MutMap[k, v, r]): Iterator[(k, v), r] \ Write(r) =
+        let MutMap(mm) = m;
+        let r = Scoped.regionOf(m);
+        Map.iterator(r, deref mm)
+
+    ///
+    /// Returns an iterator over key-value pairs in `m` zipped with the indices of the elements.
+    ///
+    pub def enumerator(m: MutMap[k, v, r]): Iterator[((k, v), Int32), r] \ Write(r) =
```
Comment (magnus-madsen) [inline main/src/library/MutMap.flix:591]: Remove this :)

**Comment by jaschdoc** [comment]
RFR

File: main/src/library/Map.flix [hunk 2]
```diff
@@ -953,10 +954,16 @@ namespace Map {
         RedBlackTree.iterator(r, t)
 
     ///
-    /// Returns an iterator over key-value pairs in `m` zipped with the indices of the elements.
+    /// Returns an iterator over values in `m`.
```
Comment (magnus-madsen) [inline main/src/library/Map.flix:957]: -> keys

File: main/src/library/MutMap.flix [hunk 3]
```diff
@@ -577,6 +577,30 @@ namespace MutMap {
...
+    ///
+    /// Returns an iterator over all key-value pairs in `m`.
+    ///
+    pub def iterator(m: MutMap[k, v, r]): Iterator[(k, v), r] \ Write(r) =
+        let MutMap(mm) = m;
+        let r = Scoped.regionOf(m);
+        Map.iterator(r, deref mm)
+
+    ///
+    /// Returns an iterator over values in `m`.
```
Comment (magnus-madsen) [inline main/src/library/MutMap.flix:589]: keys

**Comment by jaschdoc** [comment]
RFR

**Comment by jaschdoc** [comment]
RFR

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**